vunit/ui/__init__.py

```diff
@@ -1032,6 +1083,19 @@ def add_json4vhdl(self):
         """
         self._builtins.add("json4vhdl")
 
+    def run_dependent(self, source_files: List[Union[str, Path]]) -> None:
```

Comment:
I think it's best if the type hint is `Iterable[...]` instead of list. That way you can pass a set (or tuple, or any iterable class) also. In the case of tsfpga I will pass a set to this function.

Using Iterable is recommended by mypy but I can't find the right page right now. There's some info here: https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#standard-duck-types

Reply:
Agree, I usually talk about lists because it makes it more understandable for those not very experienced with Python. However, there is not reason for why we should limit mypy that way when the code works with iterables.